backend/compact-connect/lambdas/license-data/tests/function/test_data_model/test_client.py

```diff
@@ -1,20 +1,90 @@
+import json
+from uuid import uuid4
+
 from moto import mock_aws
 from tests.function import TstFunction
 
 
 @mock_aws
 class TestClient(TstFunction):
-    def test_get_ssn(self):
+    def test_get_provider_id(self):
+        from data_model.client import DataClient
+        from data_model.schema.license import LicensePostSchema, LicenseRecordSchema
+        from data_model.schema.privilege import PrivilegePostSchema, PrivilegeRecordSchema
+        from exceptions import CCNotFoundException, CCInternalException
+
+        with open('tests/resources/api/license-post.json', 'r') as f:
+            license_data = LicensePostSchema().load({
+                'compact': 'aslp',
+                'jurisdiction': 'co',
+                **json.load(f)
+            })
+
+        with open('tests/resources/dynamo/license.json', 'r') as f:
+            provider_id = json.load(f)['providerId']
+
+        self._table.put_item(
+            # We'll use the schema/serializer to populate index fields for us
+            Item=LicenseRecordSchema().dump({
+                'providerId': provider_id,
+                'compact': 'aslp',
+                'jurisdiction': 'co',
+                **license_data
+            })
+        )
+
+        with open('tests/resources/api/privilege.json', 'r') as f:
+            privilege = PrivilegePostSchema().loads(f.read())
+
+        self._table.put_item(
+            Item=PrivilegeRecordSchema().dump({
+                'providerId': provider_id,
+                'compact': 'aslp',
+                'jurisdiction': 'co',
+                **privilege
+            })
+        )
+
+        client = DataClient(self.config)
+
+        resp = client.get_provider_id(ssn='123-12-1234')  # pylint: disable=missing-kwoa
+        self.assertEqual(provider_id, resp)
+
+        with self.assertRaises(CCNotFoundException):
+            client.get_provider_id(ssn='321-21-4321')  # pylint: disable=missing-kwoa
+
+        # Associate a second provider with the same ssn - force a data consistency error
+        self._table.put_item(
+            # We'll use the schema/serializer to populate index fields for us
+            Item=LicenseRecordSchema().dump({
+                'providerId': str(uuid4()),
+                'compact': 'aslp',
+                'jurisdiction': 'co',
+                **license_data
+            })
+        )
+        with self.assertRaises(CCInternalException):
+            client.get_provider_id(ssn='123-12-1234')  # pylint: disable=missing-kwoa
```

Comment:
It's not a great reason, really. I just recognized that most of the first test would need to be replicated for the data consistency error, so opted to tack it on to the end, rather than duplicate so much code. I can separate.